### PR TITLE
Update exception messages on impersonation failure

### PR DESCRIPTION
**Pull request by apc999**
- Put more details to explain impersonation failure with a link to documentation on impersonation
- fix a wrong URL to `ALLUXIO_DEBUG_DOCS_URL`

### CODE REVIEW

File: core/common/src/main/java/alluxio/security/authentication/ImpersonationAuthenticator.java [hunk 1]
```diff
@@ -97,8 +97,13 @@ public void authenticate(String connectionUser, String impersonationUser)
     Set<String> allowedGroups = mImpersonationGroups.get(connectionUser);
 
     if (allowedUsers == null && allowedGroups == null) {
-      throw new AuthenticationException(ExceptionMessage.IMPERSONATION_NOT_CONFIGURED
-          .getMessage(connectionUser, impersonationUser));
+      throw new AuthenticationException(String.format(
+          "Failed to authenticate client user=\"%s\" connecting to Alluxio master and "
```
Comment (gpang) [inline core/common/src/main/java/alluxio/security/authentication/ImpersonationAuthenticator.java]: I think technically this exception can happen on any alluxio server, master or worker.

File: core/common/src/main/java/alluxio/security/authentication/ImpersonationAuthenticator.java [hunk 2]
```diff
@@ -123,11 +128,21 @@ public void authenticate(String connectionUser, String impersonationUser)
           }
         }
       } catch (IOException e) {
-        throw new AuthenticationException(ExceptionMessage.IMPERSONATION_GROUPS_FAILED
-            .getMessage(impersonationUser, connectionUser), e);
+        throw new AuthenticationException(String.format(
+            "Failed to authenticate client user=\"%s\" connecting to Alluxio master and "
+                + "impersonating as impersonationUser=\"%s\" to access Alluxio file system. "
+                + "Failed to get the allowed impersonation groups for impersonationUser=\"%s\"."
```
Comment (gpang) [inline core/common/src/main/java/alluxio/security/authentication/ImpersonationAuthenticator.java]: This is not the actual issue here. The issue is that alluxio cannot retrieve the groups that the impersonation user is a part of. It is not actually related to any impersonation configuration.

File: core/common/src/main/java/alluxio/security/authentication/ImpersonationAuthenticator.java [hunk 3]
```diff
@@ -123,11 +128,21 @@ public void authenticate(String connectionUser, String impersonationUser)
...
+            connectionUser, impersonationUser, impersonationUser,
+            RuntimeConstants.ALLUXIO_SECURITY_DOCS_URL), e);
       }
     }
-    throw new AuthenticationException(
-        ExceptionMessage.IMPERSONATION_DENIED.getMessage(connectionUser, impersonationUser));
+    throw new AuthenticationException(String.format(
+        "Failed to authenticate client user=\"%s\" connecting to Alluxio master and "
+            + "impersonating as impersonationUser=\"%s\" to access Alluxio file system. "
+            + "user=\"%s\" is not configured to allow as impersonationUser=\"%s\"."
```
Comment (gpang) [inline core/common/src/main/java/alluxio/security/authentication/ImpersonationAuthenticator.java]:
```suggestion
            + "user=\"%s\" is not configured to impersonate as impersonationUser=\"%s\"."
```

File: core/common/src/main/java/alluxio/security/authentication/ImpersonationAuthenticator.java (hunk 1 above)
Comment (apc999) [reply core/common/src/main/java/alluxio/security/authentication/ImpersonationAuthenticator.java]: oh? do we also authenticate clients on workers?

**Comment by apc999** [comment]
@gpang  PTAL

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Comment by apc999** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please

**Pull request merged**